src/libraries/Common/src/System/Security/Cryptography/DSAAndroid.cs

```diff
@@ -0,0 +1,423 @@
+                VerifySignatureCore(hash, signature, DSASignatureFormat.IeeeP1363FixedFieldConcatenation);
+
+            protected override bool VerifySignatureCore(
+                ReadOnlySpan<byte> hash,
+                ReadOnlySpan<byte> signature,
+                DSASignatureFormat signatureFormat)
+#else
+            public override bool VerifySignature(ReadOnlySpan<byte> hash, ReadOnlySpan<byte> signature)
+#endif
+            {
+                SafeDsaHandle key = GetKey();
+
+#if INTERNAL_ASYMMETRIC_IMPLEMENTATIONS
+                if (signatureFormat == DSASignatureFormat.IeeeP1363FixedFieldConcatenation)
+                {
+#endif
+                    int expectedSignatureBytes = Interop.AndroidCrypto.DsaSignatureFieldSize(key) * 2;
+                    if (signature.Length != expectedSignatureBytes)
+                    {
+                        // The input isn't of the right length (assuming no DER), so we can't sensibly re-encode it with DER.
+                        return false;
+                    }
+
+                    signature = AsymmetricAlgorithmHelpers.ConvertIeee1363ToDer(signature);
+#if INTERNAL_ASYMMETRIC_IMPLEMENTATIONS
+                }
+                else if (signatureFormat != DSASignatureFormat.Rfc3279DerSequence)
+                {
+                    Debug.Fail($"Missing internal implementation handler for signature format {signatureFormat}");
+                    throw new CryptographicException(
+                        SR.Cryptography_UnknownSignatureFormat,
+                        signatureFormat.ToString());
+                }
+#endif
+                return Interop.AndroidCrypto.DsaVerify(key, hash, signature);
+            }
+
+            private void ThrowIfDisposed()
+            {
+                if (_key == null)
+                {
+                    throw new ObjectDisposedException(
+#if INTERNAL_ASYMMETRIC_IMPLEMENTATIONS
+                        nameof(DSA)
+#else
+                        nameof(DSAAndroid)
+#endif
+                    );
+                }
+            }
+
+            private SafeDsaHandle GetKey()
+            {
+                ThrowIfDisposed();
+
+                SafeDsaHandle key = _key.Value;
+                CheckInvalidKey(key);
+
+                return key;
+            }
+
+            private void SetKey(SafeDsaHandle newKey)
+            {
+                // Do not call ThrowIfDisposed here, as it breaks the SafeEvpPKey ctor
+
+                // Use ForceSet instead of the property setter to ensure that LegalKeySizes doesn't interfere
+                // with the already loaded key.
+                ForceSetKeySize(BitsPerByte * Interop.AndroidCrypto.DsaKeySize(newKey));
+
+                _key = new Lazy<SafeDsaHandle>(newKey);
+            }
+
+            private static readonly KeySizes[] s_legalKeySizes = new KeySizes[] { new KeySizes(minSize: 512, maxSize: 3072, skipSize: 64) };
```

Comment:
In the tests you've commented that Android doesn't support keys less than 1024, and that it further only supports sizes that are multiples of 1024...

```suggestion
            private static readonly KeySizes[] s_legalKeySizes = new KeySizes[] { new KeySizes(minSize: 1024, maxSize: 3072, skipSize: 1024) };
```